pandas/core/strings.py

```diff
@@ -464,11 +464,9 @@ def rep(x, r):
         return result
 
 
-def str_match(arr, pat, case=True, flags=0, na=np.nan, as_indexer=False):
+def str_match(arr, pat, case=True, flags=0, na=np.nan, as_indexer=None):
```

Comment:
No, this keyword needs to stay, because it was how people could specify the 'new' behaviour before (although we said we would change this in 0.14, we never did). 
So all people still using `match` are probably specifying this keyword, AFAIU. 

See the removed warning from the documentation in the diff for some context.

In principle we could make it a FutureWarning instead of UserWarning, so we can remove it later on.